src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.ts

```diff
@@ -449,6 +385,59 @@ export class DataChartComponent implements OnInit, OnDestroy {
     }
   }
 
-}
+   // use getter setter to define the property
+   get globalFilter(): any {
+    return this._globalFilter;
+  }
+
+  @Input()
+  set globalFilter(val: any) {
+    if(val){
+
+      this.chartData = val.chartData;
+      if(val.filters){
+        this.chartData['selectedFilters'] = { };
+      }else {
+        this.chartData['selectedFilters'] = val.filters;
+      }
+     this.cdr.detectChanges();
+      this.getDataSetValue(val.chartData);
+      this.resetForm();
+     
+    }
+  }
+
+  public filterChanged(data: any):void {
+    this.chartType = data.chartType;
+    this.cdr.detectChanges();
 
+    this.currentFilters = data.filters;
+    if(data.filters){
+      this.chartData['selectedFilters'] = data.filters;
+    }else {
+      this.chartData['selectedFilters'] = {};
+    }
+    this.getDataSetValue(data.chartData);
+  }
+  public graphStatsChange(data:any):void {
+    this.showStats=data;
+  }
+  changeChartType(chartType) {
+    this.chartType = _.lowerCase(chartType);
+  }
+  showFilterPopup(){
```

Comment:
use a single method and pass the boolean value as the parameter.

Reply:
resolved